plugin/cmd.go

```diff
@@ -18,40 +19,75 @@ const (
 	PluginMagicCookieKey       = "INGEST_PLUGIN"
 )
 
-func RunPluginServer(s Source, d Destination, g prometheus.Gatherer) {
+type Option func(c *configuration)
+
+func WithLogger(l hclog.Logger) Option {
+	return func(c *configuration) {
+		c.l = l
+	}
+}
+
+func WithGatherer(g prometheus.Gatherer) Option {
+	return func(c *configuration) {
+		c.g = g
+	}
+}
+
+type configuration struct {
+	g prometheus.Gatherer
+	l hclog.Logger
+}
+
+func RunPluginServer(s Source, d Destination, opts ...Option) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
+	reg := prometheus.NewRegistry()
+
+	reg.MustRegister(
+		collectors.NewGoCollector(),
+		collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}),
+	)
```

Comment:
They are here https://github.com/connylabs/ingest/pull/78/files#r1174921354

Reply:
ok so using WithGatherer is really just in case you want to add custom labels to all of the metrics collected by the plugin, or something like that. Otherwise all metrics registration is done manually in the implementation of the destination/source

Reply:
no, WithGatherer is used when ever a plugin wants to expose metrics that are not standard. Or you don't want to expose any metrics, because by default we expose go collector and process collector. Those can be overwritten with WithGatherer.

So the developer must also manually register the go collector and process collector if they want to use their own registry plus the go collector and process collector.